src/codegen/codegen_ispc_visitor.cpp

```diff
@@ -115,7 +115,7 @@ std::string CodegenIspcVisitor::double_to_string(double value) {
         return "{:.1f}d"_format(value);
     }
     if ((value <= 1.0) && (value >= -1.0)) {
-        return "{:f}d"_format(value);
+        return "{:.16f}d"_format(value);
```

Comment:
I am thinking whether we should use the same precision for the rest of the float/double strings where `:f` is used to make sure we have the best precision and avoid any other possible issues. 
Especially in `float_to_string` in the same case. We should just use `:.6f` there.

Reply:
I don't remember/know all implications. I will let @ohm314 to provide suggestion.
(the goal would be to retain representation close to what is written in MOD file)

Reply:
Ok, for now lets change as Ioannis suggested and then we can merge this.

Reply:
So, it should be really only here and `float_to_string` where this formatting is needed.. that's good. The problem I see with always printing all digits is that sometimes you will then get at the the last digits slightly wrong representation.. but it's true that overall this approach should lead to less surprises. So I agree with @iomaganaris